raco/backends/myria/myria.py

```diff
@@ -2043,23 +2063,20 @@ def opt_rules(self, **kwargs):
 
         compile_grps_sequence = [
             myriafy,
-            [DoUntilConvergence(), RemoveEmptyFilter(), ReplaceStoreIDB()],
             [AddAppendTemp()],
-            break_communication
+            break_communication,
+            idb_until_convergence(),
         ]
 
         if kwargs.get('add_splits', True):
             compile_grps_sequence.append([InsertSplit()])
         # Even when false, plans may already include (manually added) Splits,
         # so we always need BreakSplit
         compile_grps_sequence.append([BreakSplit()])
-        compile_grps_sequence.append([RemoveSingleSplit()])
 
-        if kwargs.get('join_pull_order'):
+        if kwargs.get('join_pull_order', 'None') is not None:
```

Comment:
Again, the default of `get()` is `None`, so not sure what you're trying to do with the literal string `'None'`.